Directory.Build.targets

```diff
@@ -7,6 +7,21 @@
     <ImportDirectoryBuildTargets>false</ImportDirectoryBuildTargets>
   </PropertyGroup>
 
+  <!--
+  When .NET gets built from source, make the SDK aware there are bootstrap packages
+  for Microsoft.NETCore.App.Runtime.<rid> and Microsoft.NETCore.App.Crossgen2.<rid>.
+  -->
+  <ItemGroup Condition=" '$(DotNetBuildFromSource)' == 'true' ">
+    <KnownFrameworkReference Update="Microsoft.NETCore.App">
+      <RuntimePackRuntimeIdentifiers>%(KnownFrameworkReference.RuntimePackRuntimeIdentifiers);$(NETCoreSdkRuntimeIdentifier)</RuntimePackRuntimeIdentifiers>
+    </KnownFrameworkReference>
+    <KnownCrossgen2Pack Update="Microsoft.NETCore.App.Crossgen2">
```

Comment:
These updates should likely be conditioned to the TFMs for which you do have assets. That's why I suggested the Target in the other issue.

Reply:
This doesn't necessarily need a target. It just needs a condition similar to https://github.com/dotnet/runtime/blob/63b994bde58646cde2872c6f875afdcbb6fd6506/eng/targetingpacks.targets#L58 to check that only the `Microsoft.NETCore.App.Crossgen2` item is updated that contains the `TargetFramework=$(NetCoreAppCurrent)` metadata.

Maybe this code should instead be moved into targetingpack.targets?

Reply:
```suggestion
    <KnownCrossgen2Pack Update="@(KnownCrossgen2Pack->WithMetadataValue('Identity', 'Microsoft.NETCore.App.Crossgen2')->WithMetadataValue('TargetFramework', '$(NetCoreAppCurrent)'))">
```

Reply:
> to the TFMs for which you do have assets

That's probably hard to calculate. Maybe just applying it to `NetCoreAppCurrent` is sufficient for source build?